src/js/elements/mapchip/mapchip.js

```diff
@@ -0,0 +1,182 @@
+        let defaultFilters = [];
+        // let primaryGroup = params.primaryGroup;
+        // let groups = params.groups.filter(g => g.name != primaryGroup)
+        // this.filterController.groups = groups; // TODO should pass in the dataFilterModel, should also reset
+        // this._filterController.setDataFilterModel(dataFilterModel);
+
+        // this.filterController.clearExtraFilters();
+
+        // let filtersToAdd = [];
+        // if (typeof params.filter !== 'undefined') {
+        //     filtersToAdd = filtersToAdd.concat(params.filter);
+        // } else {
+        //     let defaultFilters = this.getDefaultFilters(params.chartConfiguration);
+        //     let nonAggFilters = this.getNonAggFilters(params.groups);
+        //     filtersToAdd = defaultFilters.concat(nonAggFilters);
+        // }
+
+        // for (let i = 1; i < filtersToAdd.length; i++) {
+        //     this.filterController.addFilter(filtersToAdd[i].default);
+        // }
+
+        // let dropdowns = $(mapChipBlockClass).find(`${filterRowClass} .mapping-options__filter`);
+
+        // this.filterController.filter = new SubindicatorFilter(
+        //     this.filterArea, groups, params.indicatorTitle, this.applyFilter, dropdowns, filtersToAdd, params.childData, '.map-options__filter-row'
+        // );
+
+        // this.filterController.setAddFilterButton();
+    }
+
+    getDefaultFilters = (chartConfiguration) => {
+        let defaultFilters = [];
+
+        if (typeof chartConfiguration.filter !== 'undefined') {
+            chartConfiguration.filter['defaults'].forEach((f) => {
+                let defaultFilter = {
+                    group: f.name,
+                    value: f.value,
+                    default: true
+                }
+
+                let item = defaultFilters.filter((df) => {
+                    return df.group === f.name
+                })[0];
+                if (item !== null && typeof item !== 'undefined') {
+                    item.default = true;
+                } else {
+                    defaultFilters.push(defaultFilter);
+                }
+            })
+        }
+
+        return defaultFilters;
+    }
+
+    getNonAggFilters = (groups) => {
+        let filterArr = [];
+
+        let nonAgg = [...groups].filter((g) => {
+            return !g.can_aggregate;
+        })
+        nonAgg.forEach((n) => {
+            let filter = {
+                group: n.name,
+                value: n.subindicators[Math.floor(Math.random() * n.subindicators.length)],
+                default: true
+            }
+
+            let item = filterArr.filter((df) => {
+                return df.group === n.name
+            })[0];
+            if (item !== null && typeof item !== 'undefined') {
+                item.default = true;
+            } else {
+                filterArr.push(filter);
+            }
+        })
+
+        return filterArr;
+    }
+
+    applyFilter = (filterResult, selectedFilter) => {
```

Comment:
This should live in the filter model - I think the right way would be to pass the filter model into the mapchip constructor

Reply:
we decided that this is good for now